assets/javascripts/discourse/components/ai-persona-editor.gjs

```diff
@@ -0,0 +1,236 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { Input } from "@ember/component";
+import { on } from "@ember/modifier";
+import { action } from "@ember/object";
+import didInsert from "@ember/render-modifiers/modifiers/did-insert";
+import didUpdate from "@ember/render-modifiers/modifiers/did-update";
+import { later } from "@ember/runloop";
+import { inject as service } from "@ember/service";
+import DButton from "discourse/components/d-button";
+import Textarea from "discourse/components/d-textarea";
+import DToggleSwitch from "discourse/components/d-toggle-switch";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+import Group from "discourse/models/group";
+import icon from "discourse-common/helpers/d-icon";
+import I18n from "discourse-i18n";
+import GroupChooser from "select-kit/components/group-chooser";
+import AiCommandSelector from "./ai-command-selector";
+
+export default class PersonaEditor extends Component {
+  @service router;
+  @service store;
+  @service dialog;
+  @service toasts;
+
+  @tracked allGroups = [];
+  @tracked isSaving = false;
+  @tracked editingModel = null;
+  @tracked showDelete = false;
+
+  @action
+  updateModel() {
+    this.editingModel = this.args.model.workingCopy();
+    this.showDelete = !this.args.model.isNew && !this.args.model.system;
+  }
+
+  @action
+  async updateAllGroups() {
+    this.allGroups = await Group.findAll();
+  }
+
+  @action
+  async save() {
+    const isNew = this.args.model.isNew;
+    this.isSaving = true;
+
+    const backupModel = this.args.model.workingCopy();
+
+    this.args.model.setProperties(this.editingModel);
+    try {
+      await this.args.model.save();
+      this.#sortPersonas();
+      if (isNew) {
+        this.args.personas.addObject(this.args.model);
+        this.router.transitionTo(
+          "adminPlugins.discourse-ai.ai-personas.show",
+          this.args.model
+        );
+      } else {
+        this.toasts.show({
```

Comment:
Sorry minor, this should be `this.toasts.success`

Reply:
will change